spec/enumerables_spec.rb

```diff
@@ -0,0 +1,202 @@
+
+describe Enumerable do
+  let(:arr_num) { (0..9).to_a }
+  let(:hash_clo) { { shoes: 5, shirt: 5, pants: 4, tie: 3, smoking: 7 } }
+  let(:arr_word) { %w[hi boys this is awesome] }
+  let(:hash_word) { { hi: 2, boys: 4, this: 4, is: 2, awesome: 7 } }
+  let(:arr_obj) { ['hello', 7, '', 0, nil] }
+  let(:arr_animals) { %w[cat rat hawk hamster] }
+  let(:arr_zero) { [].fill(0, 0..9) }
+  let(:arr_false) { [].fill(false, 0..9) }
+  let(:arr_empty) { [nil, false, [], 0, ''] }
+  let(:arr_types) { [1, 'hi', :prefix, 2.345, :count, true, nil, ['string'], :incr] }
+
+  context '#my_each' do
+    it 'Prints every value of the array' do
+      expected = ''
+      arr_num.each { |v| expected << "#{v}\n" }
+      expect { arr_num.my_each { |v| p v } }.to output(expected).to_stdout
+    end
+
+    it 'Prints every key-value pair' do
+      expected = ''
+      hash_clo.each { |v| expected << "#{v}\n" }
+      expect { hash_clo.my_each { |v| p v } }.to output(expected).to_stdout
+    end
+  end
+
+  context '#my_each_with_index' do
+    it 'Prints every value and index of the array' do
+      expected = ''
+      arr_num.each_with_index { |v, i| expected << "#{i}: #{v}\n" }
+      expect { arr_num.my_each_with_index { |v, i| puts "#{i}: #{v}" } }.to output(expected).to_stdout
+    end
+
+    it 'Prints every key-value pair and index' do
+      expected = ''
+      hash_clo.each_with_index { |h, i| expected << "#{i}: #{h}\n" }
+      expect { hash_clo.my_each_with_index { |h, i| puts "#{i}: #{h}" } }.to output(expected).to_stdout
+    end
+  end
+
+  context '#my_select' do
+    it 'Returns every value that is bigger than 3' do
+      expect(arr_num.my_select { |v| v > 3 }).to eql(arr_num.select { |v| v > 3 })
+    end
+
+    it 'Returns every value that includes an \'i\'' do
+      expect(arr_word.my_select { |v| /i/.match?(v) }).to eql(arr_word.select { |v| /i/.match?(v) })
+    end
+
+    it 'Returns every key-value pair where its key includes an \'i\'' do
+      expect(hash_word.my_select { |k, _v| /i/.match?(k) }).to eql(hash_word.select { |k, _v| /i/.match?(k) })
+    end
+
+    it 'Returns every key-value pair where its value is bigger than 4' do
+      expect(hash_word.my_select { |_k, v| v >= 4 }).to eql(hash_word.select { |_k, v| v >= 4 })
+    end
+
+    it 'Returns every key-value pair where its key includes and \'o\' and its value is bigger than 4' do
+      eval = proc { |k, v| /o/.match?(k) && (v >= 4) }
+      expect(hash_word.my_select(&eval)).to eql(hash_word.select(&eval))
+    end
+  end
+
+  context '#my_all?' do
+    it 'Should return true since all the values are lower than 10' do
+      expect(arr_num.my_all? { |v| v < 10 }).to eql(arr_num.all? { |v| v < 10 })
+    end
+
+    it 'Should return false since not all the elements are truthy' do
+      expect(arr_obj.my_all?).to eql(arr_obj.all?)
+    end
+
+    it 'Should return true since all the elements are zero' do
+      expect(arr_zero.my_all?(0)).to eql(arr_zero.all?(0))
+    end
```

Comment:
I think examples can be more descriptive. You can add examples of what type of pattern is been passed instead. Consider doing this for all occurrences.

Reply:
All the example's descriptions were changed to explain method functionality.